src/sentry/tasks/releasemonitor.py

```diff
@@ -28,51 +28,47 @@ def monitor_release_adoption(**kwargs):
-        #             dataset="sessions",
-        #             match=Entity("org_sessions"),
-        #             select=[
-        #                 Column("org_id"),
-        #                 Column("project_id"),
-        #             ],
-        #             groupby=[Column("org_id"),Column("project_id")],
-        #             where=[
-        #                 Condition(
-        #                     Column("started"), Op.GTE, datetime.utcnow() - timedelta(hours=6)
-        #                 ),
-        #                 Condition(Column("started"), Op.LT, datetime.utcnow()),
-        #             ],
-        #             granularity=Granularity(3600),
-        #             orderby=[OrderBy(Column("org_id"), Direction.ASC)],
-        #         ).set_limit(CHUNK_SIZE+1).set_offset(offset)
-        #         data = snuba.raw_snql_query(query, referrer="tasks.monitor_release_adoption")["data"]
-        #         count = len(data)
-        #         more_results = count >= CHUNK_SIZE
-        #         offset += count
-
-        #         for row in data:
-        #             aggregated_projects[row['org_id']].append(row['project_id'])
-
-        #         if not more_results:
-        #             break
-
-        #     else:
-        #         logger.info(
-        #             "monitor_release_adoption.loop_timeout",
-        #             extra={"offset": offset},
-        #         )
-        # NOTE: Hardcoded data for sentry org and sentry project for early release, in the same format snuba should return
-        aggregated_projects = {
-            1: [1],
-            315582: [5245995],
-            307710: [5227327],
-            34825: [1769835],
-        }  # org id: [project ids]
+        aggregated_projects = defaultdict(list)
+        start_time = time.time()
+        offset = 0
+        while (time.time() - start_time) < MAX_SECONDS:
+            query = (
+                Query(
+                    dataset="sessions",
+                    match=Entity("org_sessions"),
+                    select=[
+                        Column("org_id"),
+                        Column("project_id"),
+                    ],
+                    groupby=[Column("org_id"), Column("project_id")],
+                    where=[
+                        Condition(
+                            Column("started"), Op.GTE, datetime.utcnow() - timedelta(hours=6)
+                        ),
+                        Condition(Column("started"), Op.LT, datetime.utcnow()),
+                    ],
+                    granularity=Granularity(3600),
+                    orderby=[OrderBy(Column("org_id"), Direction.ASC)],
+                )
+                .set_limit(CHUNK_SIZE + 1)
+                .set_offset(offset)
+            )
+            data = snuba.raw_snql_query(query, referrer="tasks.monitor_release_adoption")["data"]
+            count = len(data)
+            more_results = count >= CHUNK_SIZE
+            offset += count
```

Comment:
I think this is off by 1? Since we're fetching `CHUNK_SIZE + 1` rows

Reply:
Hmm so we're definitely processing `CHUNK_SIZE + 1` rows each time (which is wrong if you expected `CHUNK_SIZE` to be processed), but leaving this as-is and having `for row in data` should mean we're not double counting or missing a row here, right?

I could change them both together (`offset += count -1` and `for row in data[:-1]` kinda thing, but then we'd be fetching the same row twice, no?

Reply:
I guess it doesn't really matter, but usually we fetch the extra row just to see if there's another page and discard it. Otherwise you're just processing pages of size `CHUNK_SIZE + 1`, and you still have no idea if the next page has data or not, so you may as well just process pages of `CHUNK_SIZE`

Reply:
Yeah, I see what you're saying. I'm happy to change this to work this way. It does feel more correct given how we're using this.